### PR TITLE
REL: set version to 1.5.0.dev0

### PR DESCRIPTION
**Pull request by tylerjereddy**
Don't merge until the `1.4.0` release notes update has been merged & we have branched `1.4.x`.

Just getting this ready now so it isn't done late at night..

### CODE REVIEW

File: doc/release/1.5.0-notes.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+documentation. There have been a number of deprecations and API changes
+in this release, which are documented below. All users are encouraged to
+upgrade to this release, as there are a large number of bug-fixes and
+optimizations. Before upgrading, we recommend that users check that
+their own code does not use deprecated SciPy functionality (to do so,
+run your code with ``python -Wd`` and check for ``DeprecationWarning`` s).
+Our development attention will now shift to bug-fix releases on the
+1.3.x branch, and on adding new features on the master branch.
+
+This release requires Python 3.5+ and NumPy 1.13.3 or greater.
```
Comment (tylerjereddy) [inline doc/release/1.5.0-notes.rst]: `3.6+`?

Comment (rgommers) [reply doc/release/1.5.0-notes.rst]: yep

**Comment by tylerjereddy** [comment]
Once #11061 is in, then this can go in. I think technically I can branch 1.4.x afterwards as long as I use the correct branch point on master.

**Comment by tylerjereddy** [comment]
Ok, I've merged the `1.4.0` release notes & branched `maintenance/1.4.x`, so I will now merge this to open SciPy for `1.5.0` development on the master branch.

**Pull request merged**